src/algebra/opposites.lean

```diff
@@ -364,3 +364,31 @@ rfl
 lemma units.coe_op_equiv_symm {R} [monoid R] (u : (units R)ᵒᵖ) :
   (units.op_equiv.symm u : Rᵒᵖ) = op (u.unop : R) :=
 rfl
+
+/-- A hom `α →* β` can equivalently be viewed as a hom `αᵒᵖ →* βᵒᵖ`. This is the action of the
+(fully faithful) `ᵒᵖ`-functor on morphisms. -/
+@[simps]
+def monoid_hom.op {α β} [mul_one_class α] [mul_one_class β] :
+  (α →* β) ≃ (αᵒᵖ →* βᵒᵖ) :=
+{ to_fun    := λ f, { to_fun := op ∘ f ∘ unop, map_one' := by simp, map_mul' := λ x y, by simp },
+  inv_fun   := λ f, { to_fun := unop ∘ f ∘ op, map_one' := by simp, map_mul' := λ x y, by simp },
+  left_inv  := λ f, by { ext, simp },
+  right_inv := λ f, by { ext, simp } }
+
+/-- The 'unopposite' of a monoid hom `αᵒᵖ →* βᵒᵖ`. Inverse to `monoid_hom.op`. -/
+@[simp] def monoid_hom.unop {α β} [mul_one_class α] [mul_one_class β] :
+  (αᵒᵖ →* βᵒᵖ) ≃ (α →* β) := monoid_hom.op.symm
+
+/-- A ring hom `α →+* β` can equivalently be viewed as a ring hom `αᵒᵖ →+* βᵒᵖ`. This is the action
+of the (fully faithful) `ᵒᵖ`-functor on morphisms. -/
+@[simps]
+def ring_hom.op {α β} [non_assoc_semiring α] [non_assoc_semiring β] :
+  (α →+* β) ≃ (αᵒᵖ →+* βᵒᵖ) :=
+{ to_fun    := λ f, { map_zero' := by simp, map_add' := by simp, ..f.to_monoid_hom.op },
+  inv_fun   := λ f, { map_zero' := by simp, map_add' := by simp, ..f.to_monoid_hom.unop },
```

Comment:
I think it's probably worth having the `add_monoid` version, just so you can write
```suggestion
{ to_fun    := λ f, { ..f.to_add_monoid_hom.op, ..f.to_monoid_hom.op },
  inv_fun   := λ f, { ..f.to_add_monoid_hom.unop, ..f.to_monoid_hom.unop },
  ```

Reply:
Good idea